primitives/src/asset.rs

```diff
@@ -0,0 +1,32 @@
+use crate::AssetId;
+
+use codec::{Decode, Encode};
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+/// Asset Pair representation for AMM trades
+/// ( asset_a, asset_b ) combination where asset_a is meant to be exchanged for asset_b
+///
+/// asset_in represents asset coming into the pool
+/// asset_out represents asset coming out of the pool
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Debug, Encode, Decode, Copy, Clone, PartialEq, Eq, Default)]
+pub struct AssetPair {
+	pub asset_in: AssetId,
```

Comment:
wouldn't just `in` be enough instead of repeating asset? this leads to awkward paths like `self.intention_b.assets.asset_in`

Reply:
True. But `in` is a keyword in rust. So, any other suggestion for the name ?

Reply:
I'd rather be verbose and clear than short and confusing. What we could have is intention_b.asset_in instead but then we wouldn't have assets as AssetPair (which I don't mind).